src/nft_utils/mintNFT.js

```diff
@@ -1,2 +1,359 @@
 import { ethers } from "ethers";
 import { useState, useEffect } from "react";
+
+//ABIs
+import Level1Completion from "../build/Level1Completion.json";
+import RelayHub from "../build/RelayHub.json";
+
+//GSN provider
+
+import { RelayProvider } from "@opengsn/provider/dist/RelayProvider";
+
+//Address
+
+import level1CompletionDeployed from "../deployedContractAddresses/Level1Completion.json";
+
+import whitelistDeployed from "../deployedContractAddresses/WhitelistPaymaster.json";
+
+import relayHubDeployed from "../rinkebyAddresses/RelayHub.json";
+
+import Loading from "../components/Loading";
+const HttpProvider = require("web3-providers-http");
+
+const MintNFT = ({ walletAddress }) => {
+  const [relayProvider, setRelayProvider] = useState("");
+  const [oneTimeAccount, setOneTimeAccount] = useState();
+
+  //txn results
+  const [proofOfTxn, setProofOfTxn] = useState();
+  const [userSubmittedAddress, setUserSubmittedAddress] = useState("");
+  const [poapTokenID, setPAOPTokenID] = useState(0);
+  const [poapTokenURI, setPAOPTokenURI] = useState("");
+  const [balanceUpdated, setBalanceUpdated] = useState(false);
+
+  //notifications
+  const [loading, setLoading] = useState(false);
+  const [relayMessage, setRelayMessage] = useState("");
+
+  //Paymaster
+
+  const [paymasterBalance, setPaymasterBalance] = useState(0);
+  const [whitelistPMAddress, setWhitelistPMAddress] = useState();
+
+  //contract Objects
+
+  const [relayHubContractSign, setRelayHubContractSign] = useState();
+  const [
+    Level1CompletionContractEphemeral,
+    setLevel1CompletionContractEphemeral,
+  ] = useState();
+
+  //error msg
+  const [errorMessage, setErrorMessage] = useState("");
+
+  //http provider
+  const [localHttpProvider, setLocalHttpProvider] = useState("");
+
+  //set provider and WhiteListPaymaster addresss
+
+  useEffect(() => {
+    initContract()
+      .then((result) => console.log(result, "init result"))
+      .catch((error) => console.log(error, "init error"));
+
+    async function initContract() {
+      // all addresses in rinkebyAddresses are contracts already deployed by OpenGSN. Just the addresses are needed.
+      // Whitelist Paymaster and Level1 are the contracts that will be deployed by us.
+
+      setWhitelistPMAddress(whitelistDeployed.address);
+
+      const gsnConfig = {
+        relayLookupWindowBlocks: 1e5,
+        relayRegistrationLookupBlocks: 1e5,
+        pastEventsQueryMaxPageSize: 2e4,
+        paymasterAddress: whitelistDeployed.address,
+        gasPrice: 0,
+      };
+
+      let httpweb3provider = new HttpProvider(
+        "https://eth-rinkeby.alchemyapi.io/v2/lUClO9NkAFshlkgvnVQD0IwrkYIRCHU_"
```

Comment:
Can you please move the URL into environment variable?

Reply:
done :)